--- /dev/null
+++ b/base/win/win_util.cc
@@ -0,0 +1,47 @@
+// Copyright 2012 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "base/win/win_util.h"
+
+#include <objbase.h>
+
+#include <strsafe.h>
+
+#include "base/check.h"
+
+namespace base {
+namespace win {
+
+bool IsUser32AndGdi32Available() {
+  static const bool is_user32_and_gdi32_available = [] {
+    // If win32k syscalls aren't disabled, then user32 and gdi32 are available.
+    PROCESS_MITIGATION_SYSTEM_CALL_DISABLE_POLICY policy = {};
+    if (::GetProcessMitigationPolicy(GetCurrentProcess(),
+                                     ProcessSystemCallDisablePolicy, &policy,
+                                     sizeof(policy))) {
+      return policy.DisallowWin32kSystemCalls == 0;
+    }
+
+    return true;
+  }();
+  return is_user32_and_gdi32_available;
+}
+
+std::wstring WStringFromGUID(const ::GUID& rguid) {
+  // This constant counts the number of characters in the formatted string,
+  // including the null termination character.
+  constexpr int kGuidStringCharacters =
+      1 + 8 + 1 + 4 + 1 + 4 + 1 + 4 + 1 + 12 + 1 + 1;
+  wchar_t guid_string[kGuidStringCharacters];
+  CHECK(SUCCEEDED(StringCchPrintfW(
+      guid_string, kGuidStringCharacters,
+      L"{%08lX-%04X-%04X-%02X%02X-%02X%02X%02X%02X%02X%02X}", rguid.Data1,
+      rguid.Data2, rguid.Data3, rguid.Data4[0], rguid.Data4[1], rguid.Data4[2],
+      rguid.Data4[3], rguid.Data4[4], rguid.Data4[5], rguid.Data4[6],
+      rguid.Data4[7])));
+  return std::wstring(guid_string, kGuidStringCharacters - 1);
+}
+
+}  // namespace win
+}  // namespace base
--- /dev/null
+++ b/base/win/win_util.h
@@ -0,0 +1,37 @@
+// Copyright 2012 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef BASE_WIN_WIN_UTIL_H_
+#define BASE_WIN_WIN_UTIL_H_
+
+#include <string>
+
+#include "base/base_export.h"
+#include "base/win/windows_types.h"
+
+namespace base {
+namespace win {
+
+// Returns true if the current process can make USER32 or GDI32 calls such as
+// CreateWindow and CreateDC. Windows 8 and above allow the kernel component
+// of these calls to be disabled (also known as win32k lockdown) which can
+// cause undefined behaviour such as crashes. This function can be used to
+// guard areas of code using these calls and provide a fallback path if
+// necessary.
+// Because they are not always needed (and not needed at all in processes that
+// have the win32k lockdown), USER32 and GDI32 are delayloaded. Attempts to
+// load them in those processes will cause a crash. Any code which uses USER32
+// or GDI32 and may run in a locked-down process MUST be guarded using this
+// method. Before the dlls were delayloaded, method calls into USER32 and GDI32
+// did not work, so adding calls to this method to guard them simply avoids
+// unnecessary method calls.
+BASE_EXPORT bool IsUser32AndGdi32Available();
+
+// Returns a string representation of |rguid|.
+BASE_EXPORT std::wstring WStringFromGUID(const ::GUID& rguid);
+
+}  // namespace win
+}  // namespace base
+
+#endif  // BASE_WIN_WIN_UTIL_H_
