--- a/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/threading/platform_thread.h
+++ b/base/allocator/partition_allocator/src/partition_alloc/partition_alloc_base/threading/platform_thread.h
@@ -11,6 +11,7 @@
 
 #include <cstddef>
 #include <iosfwd>
+#include <string>
 
 #include "partition_alloc/build_config.h"
 #include "partition_alloc/partition_alloc_base/component_export.h"
--- a/base/strings/string_util_impl_helpers.h
+++ b/base/strings/string_util_impl_helpers.h
@@ -16,6 +16,7 @@
 #include "base/check_op.h"
 #include "base/compiler_specific.h"
 #include "base/logging.h"
+#include "base/strings/utf_ostream_operators.h"
 #include "base/third_party/icu/icu_utf.h"
 
 namespace base::internal {
--- a/base/win/sid_unittest.cc
+++ b/base/win/sid_unittest.cc
@@ -18,6 +18,7 @@
 #include <algorithm>
 #include <optional>
 
+#include "base/check.h"
 #include "base/win/atl.h"
 #include "base/win/scoped_handle.h"
 #include "base/win/scoped_localalloc.h"
