--- a/base/memory/raw_ptr_exclusion.h
+++ b/base/memory/raw_ptr_exclusion.h
@@ -5,9 +5,8 @@
 #ifndef BASE_MEMORY_RAW_PTR_EXCLUSION_H_
 #define BASE_MEMORY_RAW_PTR_EXCLUSION_H_
 
-// Although `raw_ptr` is part of the standalone PA distribution, it is
-// easier to use the shorter path in `//base/memory`. We retain this
-// facade header for ease of typing.
-#include "partition_alloc/pointers/raw_ptr_exclusion.h"  // IWYU pragma: export
+#ifndef RAW_PTR_EXCLUSION
+#define RAW_PTR_EXCLUSION
+#endif
 
 #endif  // BASE_MEMORY_RAW_PTR_EXCLUSION_H_
