--- a/base/strings/string_util_impl_helpers.h
+++ b/base/strings/string_util_impl_helpers.h
@@ -16,7 +16,6 @@
 #include "base/check_op.h"
 #include "base/compiler_specific.h"
 #include "base/logging.h"
-#include "base/notreached.h"
 #include "base/third_party/icu/icu_utf.h"
 
 namespace base::internal {
--- a/base/strings/sys_string_conversions_apple.mm
+++ b/base/strings/sys_string_conversions_apple.mm
@@ -11,7 +11,6 @@
 #include <vector>
 
 #include "base/apple/bridging.h"
-#include "base/apple/foundation_util.h"
 #include "base/apple/scoped_cftyperef.h"
 #include "base/numerics/safe_conversions.h"
 
--- a/base/types/expected_unittest.cc
+++ b/base/types/expected_unittest.cc
@@ -12,8 +12,6 @@
 #include "base/containers/contains.h"
 #include "base/strings/to_string.h"
 #include "base/test/gmock_expected_support.h"
-#include "base/test/gtest_util.h"
-#include "base/types/strong_alias.h"
 #include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
--- a/base/win/access_token_unittest.cc
+++ b/base/win/access_token_unittest.cc
@@ -15,7 +15,6 @@
 #include "base/win/atl.h"
 #include "base/win/scoped_handle.h"
 #include "base/win/security_util.h"
-#include "base/win/windows_version.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 namespace base::win {
--- a/base/win/sid.cc
+++ b/base/win/sid.cc
@@ -27,7 +27,6 @@
 #include "base/strings/string_util_win.h"
 #include "base/win/scoped_handle.h"
 #include "base/win/scoped_localalloc.h"
-#include "base/win/windows_version.h"
 #include "openssl/sha.h"
 
 namespace base::win {
--- a/base/win/sid_unittest.cc
+++ b/base/win/sid_unittest.cc
@@ -22,7 +22,6 @@
 #include "base/win/scoped_handle.h"
 #include "base/win/scoped_localalloc.h"
 #include "base/win/win_util.h"
-#include "build/branding_buildflags.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 namespace base::win {
