--- a/base/strings/string_util_impl_helpers.h
+++ b/base/strings/string_util_impl_helpers.h
@@ -16,7 +16,6 @@
 #include "base/check_op.h"
 #include "base/compiler_specific.h"
 #include "base/logging.h"
-#include "base/notreached.h"
 #include "base/third_party/icu/icu_utf.h"
 
 namespace base::internal {
--- a/base/types/expected_unittest.cc
+++ b/base/types/expected_unittest.cc
@@ -12,8 +12,6 @@
 #include "base/containers/contains.h"
 #include "base/strings/to_string.h"
 #include "base/test/gmock_expected_support.h"
-#include "base/test/gtest_util.h"
-#include "base/types/strong_alias.h"
 #include "testing/gmock/include/gmock/gmock.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
