--- a/base/win/atl_throw.cc
+++ b/base/win/atl_throw.cc
@@ -9,16 +9,17 @@
 #include "base/compiler_specific.h"
 #include "base/debug/alias.h"
 #include "base/immediate_crash.h"
-#include "base/process/memory.h"
 
 namespace base {
 namespace win {
 
 NOINLINE void __stdcall AtlThrowImpl(HRESULT hr) {
   base::debug::Alias(&hr);
+#if 0
   if (hr == E_OUTOFMEMORY) {
     base::TerminateBecauseOutOfMemory(0);
   }
+#endif
   ImmediateCrash();
 }
 
